packages/origin-ui-core/src/components/Account/UserProfile/UserEmailChange.tsx

```diff
@@ -0,0 +1,153 @@
+    }
+
+    const INITIAL_VALUES = {
+        email: user.email
+    };
+
+    async function ValidationHandler(values: typeof INITIAL_VALUES) {
+        try {
+            await VALIDATION_SCHEMA.validate(values, {
+                abortEarly: false
+            });
+            return {};
+        } catch (err) {
+            return yupToFormErrors(err);
+        }
+    }
+
+    return (
+        <Formik
+            initialValues={INITIAL_VALUES}
+            validateOnMount={true}
+            onSubmit={submitForm}
+            validate={ValidationHandler}
+        >
+            {(formikProps: FormikProps<typeof INITIAL_VALUES>) => {
+                const { isSubmitting, touched, setValues } = formikProps;
+                const fieldDisabled = isSubmitting || !isEditing;
+
+                return (
+                    <Form translate="no">
+                        <Paper className={classes.container}>
+                            <Typography variant="h5">{t('user.profile.subtitle.email')}</Typography>
+                            <Grid container spacing={3}>
+                                <Grid item xs={6}>
+                                    <FormInput
+                                        label={t('user.properties.email')}
+                                        property="email"
+                                        disabled={fieldDisabled}
+                                        className="mt-3"
+                                        required
+                                    />
+                                </Grid>
+                            </Grid>
+                            {isEditing && (
+                                <Button
+                                    style={{ marginRight: 10 }}
+                                    type="button"
+                                    variant="contained"
+                                    color="primary"
+                                    className="mt-3 right"
+                                    disabled={!touched.email}
+                                    onClick={async () => {
+                                        await formikProps.validateForm();
+                                        await formikProps.submitForm();
+                                        setIsEditing(false);
+                                        dispatch(clearAuthenticationToken());
+                                    }}
+                                >
+                                    {t('user.actions.save')}
+                                </Button>
+                            )}
+                            {isEditing && (
+                                <Button
+                                    type="button"
+                                    variant="contained"
+                                    color="primary"
+                                    className="mt-3 right"
+                                    onClick={() => {
```

Comment:
When changing email and then clicking Cancel - Save button is no longer disabled.

Reply:
fixed, now should be disabled after re-entering editing mode